arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -815,23 +1133,31 @@ void RocksDBMetaCollection::bufferUpdates(rocksdb::SequenceNumber seq,
   if (!_logicalCollection.useSyncByRevision()) {
     return;
   }
+  
+  TRI_ASSERT(!inserts.empty() || !removals.empty());
+  
+  std::unique_lock<std::mutex> guard(_revisionTreeLock);
+
+  if (_revisionTreeApplied >= seq) {
+    LOG_TOPIC("1fe4d", TRACE, Logger::ENGINES)
+        << "rejecting change with too low sequence number " << seq << " for collection " << _logicalCollection.name();
```

Comment:
Shouldn't this be at least a warning? After all, this situation should never occur.